Update wfc3 notebook dependencies

haticekaratay opened the pull request:
This PR is to update dependencies from their pinned versions. The notebooks tested locally with Python=3.12 as validation in CI runs on Python=3.12.

sosey commented on notebooks/WFC3/calwf3_v1.0_cte/pre-requirements.sh:
Is there a reason that hstcal is being installed from the older ssb server and not conda-forge?
https://anaconda.org/conda-forge/hstcal

hstcal on the ssb server is at  2.7.1 and conda-forge has 3.0.2

haticekaratay replied:
No special reason. Good catch! I will update to conda-forge in the upcoming commits.

bjkuhn replied:
The `calwf3_v1.0_cte` notebook actually does need the old version of hstcal.  This is because we're showing users how to access the older v1.0 CTE correction, which was available in calwf3 versions 3.3 to 3.5.2.  I have specifically pinned hstcal v2.5.0 because it provides access to calwf3 v3.5.2, the most recent version to still include the v1.0 CTE correction.  I don't believe hstcal version 2.5.0 is hosted on conda-forge so we therefore need to continue using the ssb server.  **Please keep hstcal pinned to v2.5.0**

sosey commented on notebooks/WFC3/ir_scattered_light_calwf3_corrections/requirements.txt:
```diff
@@ -1,9 +1,10 @@
+astropy>=5.3.3
+astroquery>=0.4.6
+drizzlepac>=3.5.1
+ginga==4.0.1  # The notebook needs ginga.util.zcale
```
does this need to be pinned or can it be a minimum?

haticekaratay replied:
I have noticed that the `Ginga. scale` API has been changed, which is why I had to pin it to this version. I will attempt to update some of the notebooks, as long as they do not involve an understanding of scientific nuances, and replace them with the `astropy.visualization.ZScaleInterval`. For these updated notebooks, I will also contact the authors to get their input.

sosey reviewed (changes requested):
Just a couple comments, esp check references to the ssb server for hstcal

haticekaratay commented:
Hi @sosey, I believe we're good to go with merging this PR. I've addressed the comments. As for the Ginga pinning, I've updated one of the notebooks by replacing the dependency with Astropy visualization Zscale – it was a simple change. However, there are three others that require a more thorough understanding of the workflows, so I'll leave those for the authors/curators to update. I've provided the list of notebooks that need updating to the authors.

FDauphin commented on notebooks/WFC3/uvis_pam_corrections/WFC3_UVIS_Pixel_Area_Map_Corrections_for_Subarrays.ipynb:
```diff
@@ -471,7 +471,7 @@
     "\n",
     "# We use zcale to find the min and max for plotting\n",
-    "vmin_diff_512, vmax_diff_512 = zscale(diff_data_512)\n",
+    "vmin_diff_512, vmax_diff_512 = z.get_limits(diff_data_512)\n",
```
Are you importing `ZScaleInterval` as `z`?

haticekaratay replied:
> Are you importing `ZScaleInterval` as `z`?

`ZScaleInterval` in the above code cells is initialized as z. I followed the convention in other notebooks.

FDauphin commented on notebooks/WFC3/uvis_pam_corrections/requirements.txt:
```diff
@@ -1,5 +1,4 @@
+astropy>=5.3.3
+astroquery>=0.4.6
+matplotlib>=3.7.0
+numpy<2.0  # drizzlepac 3.7.0.1 requires numpy<2.0, but you have numpy 2.0.1 which is incompatible
```
`drizzlepac` is not installed so why restrict `numpy`?

FDauphin reviewed (changes requested):
@haticekaratay thank you for working on this. The only request change is to maintain `hstcal==2.5.0` for `calwf3_v1.0_cte` as that version is necessary for the notebook. In addition, that notebook also uses `photutils` and currently there's no versions on the `requirements.txt` so below are what I recommend:
```
astropy>=5.3.3
astroquery>=0.4.6
crds>=11.17.9
matplotlib>=3.7.0
numpy>=1.26.4
photutils==1.12.0
scipy>=1.14.0
wfc3tools>=1.4.0
```

haticekaratay commented:
@mgough-970, I noticed a [new error](https://github.com/spacetelescope/hst_notebooks/actions/runs/10583170299/job/29324591605?pr=299) during the installation process that did not occur previously. Should I delay resolving the dependencies until your work on the CI is finished?

```CondaValueError: You have chosen a non-default solver backend (libmamba) but it was not recognized. Choose one of: classic```

haticekaratay commented on notebooks/WFC3/calwf3_v1.0_cte/pre-requirements.sh:
```diff
@@ -1 +1 @@
-conda install --yes -c http://ssb.stsci.edu/astroconda hstcal==2.5.0
```
> An older version of HSTCAL is required for the notebook so this doesn't need to be updated.

In such cases, should this notebook be rewritten or perhaps deprecated? What do you think about this, @sosey?

FDauphin replied:
Apologies, I should have been more specific. The purpose of the notebook is to showcase a particular advantage `hstcal==2.5.0` has over recent versions. So it's not that the notebook can't run without the recent version (it will), it's that the notebook needs the older version to showcase that advantage. Does that help clarify things?

FDauphin reviewed (approved):
LGTM; thanks for working on this 🙌🏾

sosey commented on notebooks/WFC3/calwf3_v1.0_cte/requirements.txt:
```diff
@@ -4,6 +4,6 @@ crds
-photutils
+photutils==1.12.0
 scipy
 wfc3tools
```
This directory only has requirements updated, but I see a pre-requirements file that has an older hstcal referenced from astroconda. I think it was just missed

haticekaratay replied:
@sosey, I reverted this for the reasons mentioned by @FDauphin. Please let me know about the consensus; I can update it accordingly.

sosey reviewed (changes requested):
I think there's just one more pre-requirements file to update

FDauphin commented on notebooks/WFC3/tvb_flattenramp/pre-requirements.sh:
```diff
@@ -1 +1 @@
-conda install --yes -c conda-forge hstcal==2.7.4
+conda install --yes -c conda-forge hstcal
```
All other notebooks excluding calwf3_v1 had that requirement by default

haticekaratay replied:
> this is the only pre-requirements file that got hstcal updated to the lastest on conda-forge. Is there a specific reason that it's not updated on the other notebooks? I'll mark them specifically in this review

Hi @sosey, all the others are up-to-date to the latest on the main, so this is the remaining one.
